pkg/check/uploadstress/uploadstress.go

```diff
@@ -0,0 +1,108 @@
+package uploadstress
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"sort"
+	"time"
+
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/check"
+	"github.com/ethersphere/beekeeper/pkg/random"
+	"golang.org/x/sync/errgroup"
+)
+
+// compile check whether UploadStress implements interface
+var _ check.Check = (*UploadStress)(nil)
+
+// UploadStress check
+type UploadStress struct{}
+
+// NewUploadStress returns new ping check
+func NewUploadStress() *UploadStress {
+	return &UploadStress{}
+}
+
+// Run executes upload stress check
+func (u *UploadStress) Run(ctx context.Context, cluster *bee.Cluster, o check.Options) (err error) {
+	buffer := 10
+
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return fmt.Errorf("node clients: %w", err)
+	}
+
+	nodeNames := []string{}
+	for k := range clients {
+		nodeNames = append(nodeNames, k)
+	}
+	sort.Strings(nodeNames)
+	nodeCount := int(math.Round(float64(len(clients)*o.UploadNodesPercentage) / 100))
+	rnd := random.PseudoGenerator(o.Seed)
+	picked, _ := randomPick(rnd, nodeNames, nodeCount)
+
+	rnds := random.PseudoGenerators(rnd.Int63(), nodeCount)
+
+	uGroup := new(errgroup.Group)
+	uSemaphore := make(chan struct{}, buffer)
+	for i, p := range picked {
+		o := o
```

Comment:
it is not needed to reassign `o` as it is always the same.

Reply:
Done.